src/test/java/spoon/support/visitor/java/JavaReflectionTreeBuilderTest.java

```diff
@@ -916,4 +918,43 @@ void testNonStaticInnerClassConstructorWithEnclosingClassArgument() throws Class
 		);
 	}
 
+	@ParameterizedTest
+	@ValueSource(strings = {
+		"class Victim {}",
+		"enum Victim {;}",
+		"interface Victim {}",
+		"@interface Victim {}"
+	})
+	void testInnerClassesAreNotAddedToPackage(String collider) throws ClassNotFoundException {
+		// contract: Inner classes are not added to their package
+		ClassLoader loader = JavacFacade.compileFiles(
+			Map.of(
+				"First.java",
+				"class First {\n"
+					+ collider +
+					"}\n",
+				"Victim.java",
+				"class Victim {\n" +
+					"  class Inner {\n" +
+					"    int bar;\n" +
+					"  }\n" +
+					"}\n"
+			),
+			List.of()
+		);
+		Factory factory = createFactory();
+		// Load the victim
+		factory.Type().get(loader.loadClass("Victim"));
+		// Let it get replaced by First$Collider
+		factory.Type().get(loader.loadClass("First"));
+
+		// This will throw if the replacement was successful
+		CtType<?> victim = factory.Type().get(loader.loadClass("Victim$Inner"));
```

Comment:
This could be wrapped into `assertDoesNotThrow` to clarify what went wrong. (also the "successful" sounds very positive, while it really isn't, but I don't have any better idea for that comment right now)